crates/btcio/src/broadcaster/state.rs

```diff
@@ -11,8 +11,8 @@ pub(crate) struct BroadcasterState {
     /// Next index from which we should next read the [`L1TxEntry`] to check and process
     pub(crate) next_idx: u64,
 
-    /// Unfinalized [`L1TxEntry`]s which the broadcaster will check for
-    pub(crate) unfinalized_entries: BTreeMap<u64, L1TxEntry>,
+    /// Unfinalized [`L1TxEntry`]s which the broadcaster will check for.
+    pub(crate) unfinalized_entries: Vec<(u64, L1TxEntry)>,
```

Comment:
Should specify that this is like a key->value relationship even though it's a vec (and what the u64 signifies).  Or create an entry type for it.

Reply:
What about using [`indexmap`](https://lib.rs/crates/indexmap)?

Reply:
@storopoli, since the entries will be processed linearly and don't need O(1) access, these can just be a `Vec`.

@delbonis added a type alias instead of a new type since this is internal to the crate.